src/main/java/de/komoot/photon/PhotonDoc.java

```diff
@@ -116,4 +126,62 @@ public boolean isUsefulForIndex() {
 
         return true;
     }
+    
+    /**
+     * Complete doc from nominatim address information.
+     */
+    public void completeFromAddress() {
+        if (address == null) return;
+
+        String addressStreet = address.get("street");
+        if (addressStreet != null) {
+            if (this.street == null) {
+                this.street = new HashMap<>();
+            }
+            setOrReplace(addressStreet, this.street, "street");
+        }
+        
+        String addressCity = address != null ? address.get("city") : null;
+        if (addressCity != null) {
+            if (this.city == null) {
+                this.city = new HashMap<>();
+            }
+            setOrReplace(addressCity, this.city, "city");
+        }
+        
+        String addressDistrict = address != null ? address.get("suburb") : null;
+        if (addressDistrict != null) {
+            if (this.district == null) {
+                this.district = new HashMap<>();
+            }
+            setOrReplace(addressDistrict, this.district, "suburb");
+        }
+        
+        String addressLocality = address != null ? address.get("neighbourhood") : null;
+        if (addressLocality != null) {
+            if (this.locality == null) {
+                this.locality = new HashMap<>();
+            }
+            setOrReplace(addressLocality, this.locality, "neighbourhood");
+        }
+        
+        String addressPostCode = address != null ? address.get("postcode") : null;
+        if (addressPostCode != null && !addressPostCode.equals(this.postcode)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Replacing postcode "+this.postcode+" with "+ addressPostCode+ " for osmId #" + osmId);
+            }
+            this.postcode = addressPostCode;
+        }
+    }
+
+    private void setOrReplace(String name, Map<String, String> namesMap, String field) {
+        String existingName = namesMap.get("name");
+        if (!name.equals(existingName)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Replacing "+ field +" name '"+existingName+"' with '"+ name+ "' for osmId #" + osmId);
+            }
+            namesMap.put("formerName", existingName);
```

Comment:
I was more thinking of adding it to the context hashset. If you have a look at `completePlace()` function in `NominatimConnector.java`, you'll see that this is what happens when multiple city-like address parts show up.